uc-cdis/cloud-automation
# Chore/dbsetup

* `gen3 testsuite` enhancements
    - remove `g3k_testsuite` 
    - introduce `--filter` and test tags - ex:
        - `gen3 testsuite --filter local`
        - `gen3 testsuite --filter secrets,db,test_shunit_assert`
    - move tests to `gen3/test/` folder
* introduce `gen3 secrets` command - `gen3 secrets sync` does the following:
    - commit outstanding changes to `$(gen3_secrets_folder)` - which currently maps to `~/$vpc_name`
    - update `X-creds` kubernetes secrets with current values from `$(gen3_secrets_folder)/creds.json`
    - update `X-g3auto` k8s secrets with current values from `$(gen3_secrets_folder)/g3auto/`
    - note: `gen3 secrets decode $secretName` is another handy function
* introduce `gen3 db` command - `gen3 db setup $serviceName` does the following:
    - verifies that the $serviceName database does not already exist
    - selects a db server from the db server farm 
    - creates a new db and new user on the selected server
    - saves the secrets for the new db as `$(gen3_secrets_folder)/g3auto/$serviceName/dbcreds.json`
    - calls `gen3 secrets sync` to sync the secrets with k8s

The ideas here:
* we can add secrets files to `$(gen3_secrets_folder)/g3auto/$serviceName/`, and the `gen3` automation automatically creates a `${serviceName}-g3auto` secret with entries for each file in the folder - we no longer need to write custom code for secret creation into `kube-setup-secrets.sh` and friends
* we can add a `gen3 db setup $serviceName" || true` call to `kube-setup-${serviceName}.sh` to allocate a new database for that service with a `${serviceName}-g3auto` k8s secret with a `dbcreds.json` entry with the configuration for connecting to the database.  `gen3 db psql $serviceName` works too.
* run unit tests (that should pass without a k8s or aws environment) locally on a laptop like this:
```
export GEN3_HOME=/path/to/cloud-automation
source $GEN3_HOME/gen3/gen3setup.sh
gen3 testsuite --filter local
```

### New Features
- `gen3 secrets`
- `gen3 db`

### Breaking Changes


### Bug Fixes


### Improvements
- `gen3 testsuite --filter`

### Dependency updates


### Deployment changes
